### ISSUE TITLE
Transect pictures from the app not visible on the website

### ISSUE THREAD

**Issue by CrisSevilleja** [BiologicalRecordsCentre/ABLE#720]
Hello John, 

I think you are the person to explain this or maybe @kazlauskis. When doing a transect on the app, I normally take pictures of the section to show the state. I realised that the pictures are not present on the website when exploring the transect survey, even there is a place for showing media and pictures. See picture below of an example where seven pictures were uploaded but not visible on the website.

https://butterfly-monitoring.net/mydata/samples/details?sample_id=26287807 

I wonder if the storage of the pictures done per section is the issue of why they are not shown on the website on the sample detail page.  Other users had the same issue.

**Comment by kazlauskis** [comment]
I can see the images in the warehouse, so the app has uploaded them correctly. I'm not sure why they're not showing up on the website though.

**Comment by johnvanbreda** [comment]
@DavidRoy this relates to the images being stored at a section level, not at the transect level, so the sample details page for a transect would need some enhancement in order to output the details of the sections, including the images.

**Comment by CrisSevilleja** [comment]
Just to add, the Hungarian coordinator also mentioned, the comments are not shown at the sample detail page. Maybe it happens the same as with the pictures.

**Comment by CrisSevilleja** [comment]
related issue #616

**Comment by DavidRoy** [comment]
@andrewvanbreda can you look into this please

**Comment by andrewvanbreda** [comment]
Hi @DavidRoy @CrisSevilleja ,

As noted by John, these images are held at the section level, but the website interface is currently a large grid containing only abundances, this causes a difficulty tow ork out where to put the images on the screen.
(actually a further complication is Hungary does something different, and they have panels with attributes).

I do note actually that Gary included an option to add a Section Level images for the data entry form.
However there is an issue, when this is enabled the whole interface for the page changes to be more like what Hungary have, with panels for each section instead of columns.
If I remember correctly from previous discussions about attributes, it is not desirable to enforce this onto all users, and we want users to have the option of simple entry.
(This is all before we even consider the need to test this, as I am not sure it has been used anywhere before).

So I think I will propose the following solution instead.

Important: the following assumes you don't want to save these images on the website, and you just want to be viewing the app ones.

1. On the Walk Data Entry page, on the grid of sections and abundances, have "View Details and Images" link for each section, possible near the top of each section column.
(Hungary would have to be at the top of each panel instead).
I have attached a screenshot of something similar to what I mean (the position of the link could be different to the screenshot, or perhaps the section name header at the top could be an active link instead if we don't want a new row).

2. This will link to a new version of this page
https://butterfly-monitoring.net/mydata/samples/details?sample_id=26287808

This is the same page that can be accessed from the "My Samples" page "View sample details" icon.
Fortunately I notice this page does accept a section sample ID without me having to change the system.

I suggest I create a newly configured version of this page which changes some things (the most obvious to remove the "Edit this sample" button which won't work for a section sample).

Doing it this way also has the advantage that you will see some other details like a map, and actually it should probably fix your missing comments issue noted earlier in this thread

Let me know what you think of this solution, particularly the limitation that section level images will only be viewable on the website.

![Image](https://github.com/user-attachments/assets/cede2915-6b8b-49c4-9be0-ccc4750338a4)

**Comment by larspett** [comment]
That sounds quite excellent, it would be really helpful to see the transect (segment, or transect with the segment highlighted) on the map /Lars

24 apr. 2025 kl. 17:43 skrev andrewvanbreda ***@***.***>:

﻿
[https://avatars.githubusercontent.com/u/2463614?s=20&v=4]andrewvanbreda left a comment (BiologicalRecordsCentre/ABLE#720)<https://github.com/BiologicalRecordsCentre/ABLE/issues/720#issuecomment-2828091983>

Hi @DavidRoy<https://github.com/DavidRoy> @CrisSevilleja<https://github.com/CrisSevilleja> ,

As noted by John, these images are held at the section level, but the website interface is currently a large grid containing only abundances, this causes a difficulty tow ork out where to put the images on the screen.
(actually a further complication is Hungary does something different, and they have panels with attributes).

I do note actually that Gary included an option to add a Section Level images for the data entry form.
However there is an issue, when this is enabled the whole interface for the page changes to be more like what Hungary have, with panels for each section instead of columns.
If I remember correctly from previous discussions about attributes, it is not desirable to enforce this onto all users, and we want users to have the option of simple entry.
(This is all before we even consider the need to test this, as I am not sure it has been used anywhere before).

So I think I will propose the following solution instead.

Important: the following assumes you don't want to save these images on the website, and you just want to be viewing the app ones.

  1.  On the Walk Data Entry page, on the grid of sections and abundances, have "View Details and Images" link for each section, possible near the top of each section column.
(Hungary would have to be at the top of each panel instead).
I have attached a screenshot of something similar to what I mean (the position of the link could be different to the screenshot, or perhaps the section name header at the top could be an active link instead if we don't want a new row).

  2.  This will link to a new version of this page
https://butterfly-monitoring.net/mydata/samples/details?sample_id=26287808

This is the same page that can be accessed from the "My Samples" page "View sample details" icon.
Fortunately I notice this page does accept a section sample ID without me having to change the system.

I suggest I create a newly configured version of this page which changes some things (the most obvious to remove the "Edit this sample" button which won't work for a section sample).

Doing it this way also has the advantage that you will see some other details like a map, and actually it should probably fix your missing comments issue noted earlier in this thread

Let me know what you think of this solution, particularly the limitation that section level images will only be viewable on the website.

View.Section.Details.png (view on web)<https://github.com/user-attachments/assets/cede2915-6b8b-49c4-9be0-ccc4750338a4>

**Comment by CrisSevilleja** [comment]
I agree, I think it is a good solution. And seeing the pictures on the website is what we are looking for. Thanks @andrewvanbreda for the suggested solution

**Comment by andrewvanbreda** [comment]
Hi @larspett @CrisSevilleja 
Thanks for the positive feedback, I will start working towards getting this put on the live site.

@DavidRoy
I assume this functionality something we will go ahead with, so I will start programming it (I don't think there are any alternative solutions at the moment). We can adjust the solution if it isn't quite to your liking.

**Comment by DavidRoy** [comment]
Thanks @andrewvanbreda - please go ahead

**Comment by andrewvanbreda** [comment]
Hi @DavidRoy @CrisSevilleja @larspett,

I have deployed an initial version of this.

There are some limitations, so I have not advertised the feature in the page instructions until it is a bit more robust, however it is there if you wish to use it (it is also compatible with Hungary's setup).
You can see the app photos and a few other details about the section sample.

Important:
At the moment there isn't a View Sections Details link.
To use the new feature, the section code itself is an active link to the section details page. 
(this is because if I called it "View Section Details", people would have certain expectations of the functionality which might not be met yet. This may change in the future)

Limitations:

- There is no map of the section and transect yet (coming soon)
- It only works when editing existing non-training walks (link is deactivated in other situations)
- It only works if there are occurrences. At the moment you will see a "sample not found" message if there are no occurrences. I could try to deactivate the link in this situation, but it would slow the data entry page down
(This is actually an existing issue https://github.com/BiologicalRecordsCentre/ABLE/issues/718 which I am looking into)

**Comment by CrisSevilleja** [comment]
Thank you @andrewvanbreda how can I see this new feature? I understand is not possible in the live website

**Comment by andrewvanbreda** [comment]
Hi @CrisSevilleja ,

It is on live.

All you need to do is go to your walks page, click on an existing walk to open the data entry page, go to the second page (which has the species grid on it).
Then you can simply click on a section code (as per the screenshot) to open the new page.

This is deliberately not too obvious at the moment because we don't want hundreds of people trying it and then getting confused when it doesn't work when there are no occurrences.
I think once that is resolved we can change it to "View Photos", and then once the map is operational it can be changed to "View Section Details" or similar

![Image](https://github.com/user-attachments/assets/f2ecc0be-c3eb-40f8-82b8-091a245379bc)

**Comment by andrewvanbreda** [comment]
Hi @DavidRoy @CrisSevilleja @larspett,

Section Details page is now much improved on live.

The Section Details link now shows all the time, and isn't limited by scenario.
The link is now clearer and says "Section details" (will respond to translation as required)
The Section Details page itself now has map showing the section route geometry correctly (I had to previously hide the map as it wasn't behaving in the required way).

The bug where section details doesn't show where there were no occurrences is also gone, this bug was fixed a couple of weeks ago.

@CrisSevilleja: Perhaps it is best if you close this issue once you are happy with it as you raised it originally (or alternatively comment further).

Hope that helps.

Andy
